.github/workflows/e2e.yml

```diff
@@ -160,22 +160,23 @@ jobs:
         working-directory: opencollective-frontend
         run: CYPRESS_INSTALL_BINARY=0 npm ci --prefer-offline --no-audit
 
+      - name: Set commit hash (frontend)
+        run: |
+          cd opencollective-frontend
+          echo "FRONTEND_COMMIT_HASH=$(git rev-parse HEAD)" >> $GITHUB_ENV
+
       - name: Restore build (frontend)
         uses: actions/cache@v1
         id: next-build
         with:
           path: opencollective-frontend/dist
-          key: ${{ runner.os }}-next-build-${{ hashFiles('opencollective-frontend/**') }}
+          key: ${{ runner.os }}-next-build-${{ env.FRONTEND_COMMIT_HASH }}
 
       - name: Restore .next cache (frontend)
-        if: steps.next-build.outputs.cache-hit != 'true'
         uses: actions/cache@v1
         with:
           path: opencollective-frontend/.next/cache
-          key: ${{ runner.os }}-next-cache-${{ hashFiles('opencollective-frontend/**') }}
-          restore-keys: |
-            ${{ runner.os }}-next-cache-${{ hashFiles('opencollective-frontend/**') }}
-            ${{ runner.os }}-next-cache-
```

Comment:
Why removing this line?

Reply:
Do you understand what it does?

Reply:
@znarf : My understanding is if the key wasn't hit, then the `restore-key` is used and but in this case both are the same which didn't make sense to me. So I was trying out removing it to see what happens. 🤔

Reply:
In case the first restore key is not found, it fall back on the partial one. This is on purpose and should not be removed if there is no explicit reason.

https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows

Reply:
(Please don't mark such ongoing discussions as resolved, it's misleading.)

Reply:
I can't accept the `restore-keys` to be removed if there is no good reason for it, it was built like this on purpose.

Reply:
@znarf : That makes sense, I've re-added the cache keys back. Will think more on the reasoning behind the addition of these fallback keys. 🤔

Reply:
@SudharakaP The reason is giving there: https://github.com/opencollective/opencollective-api/pull/6274#discussion_r669380270